pkg/gui/controllers/branches_controller.go

```diff
@@ -375,7 +375,7 @@ func (self *BranchesController) createPullRequestMenu(selectedBranch *models.Bra
 				OnPress: func() error {
```

Comment:
I noticed that when using the "against selected branch" option for a GitHub fork, GitHub won't automatically open the PR against the upstream. But that has nothing to do with my change (same behaviour on current `master`). Might be worth an additional look.

Reply:
I had another look at this: Apparently, to create a PR against a different remote (mainly the upstream I'd guess) the URL would need to look different.

Instead of something like

```
https://github.com/moha-gh/lazygit/compare/master...mh%2Fpr-suggest-remote-branches-only?expand=1
```

one would need to use

```
https://github.com/jesseduffield/lazygit/compare/master...moha-gh:lazygit:mh/pr-suggest-remote-branches-only?expand=1
```

That means the "repo URL" of the target remote (`https://github.com/jesseduffield/lazygit/` instead of ` https://github.com/moha-gh/lazygit`) is used and the full name of the source repo is prepended to the remote branch branch (with `/` replaced by `:`).

To handle this correctly, the `createPullRequest()` would need to be extended I think - so one can pass target and source remote as well, allowing the function to build the URL correctly.

Reply:
That makes sense to me. I'm happy for that to be handled as a separate pull request (given that it's currently broken on master). As you say, it'll require updating `HostHelper.GetPullRequestURL` to take extra arguments.

Reply:
Ok, then let's fix this separately :+1: